Inscription api

Fcornaire opened the pull request:
Mise en place de l'inscription

Github #2  #10  #9

Commit:
<feat> Inscription [Test]
-Ajout des tests de création d'utilisateur et get d'un utilisateur
-MAJ POM pour les dépendances de test

Commit:
<chore> Inscription [Securité]
-Début de mise en place des appels sécurisés (autorisation d'appel que sur /api/user en POST et blocage du reste par exemple)
-Encryptage du mot de passe de la création d'un utilisateur
-Maj doc (création d'utilisateur)
-Maj POM (dépendence pour la sécurité)

Commit:
<chore> API [Sécurité]
-Mise en place de la sécurisation des appels vers l'API (Token JWT,filtrage des urls,vérification en base)
-Mise à jour de la data FYTCUser pour prendre en compte la sécurité (field enable and role)
-Mise à jour du controller FYTCUserController pour attendre du JSON en entrée de createUser (au lieu des paramètres d'url)
-Ajout de la dépendence JWT dans le pom API

Commit:
<chore> Environnement
-Mise en place de l'environnement de prod (prod properties ) et d'intégration (int properties)
-Ajout test Controller FYTC (oubli)

Commit:
<chore> Dockerfile
-Création des Dockerfile Intégration et Test

MawsFr commented on findyourtrashcan-api/src/main/java/fr/femm/findyourtrashcan/controller/FYTCUserController.java:
```diff
@@ -27,13 +27,9 @@
-		FYTCUser user = new FYTCUser(username,password,mail);
-		
+	public FYTCUser createUser(@RequestBody FYTCUser user) {		
+		logger.info("WebService createUser [username : "+ user.getUsername() +", password : "+ user.getPassword() +", mail : "+ user.getEmail() +"]");
```
Formater le code pour avoir des espace entre les "+" (ctrl + shift + f)
cf bonne pratique

MawsFr commented on findyourtrashcan-api/src/main/java/fr/femm/findyourtrashcan/controller/FYTCUserController.java:
```diff
@@ -27,13 +27,9 @@
-		FYTCUser user = new FYTCUser(username,password,mail);
-		
+	public FYTCUser createUser(@RequestBody FYTCUser user) {		
+		logger.info("WebService createUser [username : "+ user.getUsername() +", password : "+ user.getPassword() +", mail : "+ user.getEmail() +"]");
```
Formater le code pour avoir des espace entre les "+" (ctrl + shift + f)
cf bonne pratique

Fcornaire replied:
oui

MawsFr replied:
Et ils sont où mes espaces xD ?

Fcornaire replied:
Ah mince,pensait l'avoir fait

MawsFr commented on findyourtrashcan-api/src/main/java/fr/femm/findyourtrashcan/data/FYTCUser.java:
```diff
@@ -33,6 +37,8 @@ public FYTCUser(String username, String password, String email) {
 		this.password = password;
 		this.email = email;
+		this.enabled = true;
+		this.role = "USER";
```
Peut être faire une classe ?

Fcornaire replied:
Bah je rajoute en tache trello,a voir pour après

MawsFr replied:
Une classe bordel xD

MawsFr commented on findyourtrashcan-api/src/main/java/fr/femm/findyourtrashcan/controller/FYTCUserController.java:
```diff
@@ -27,13 +27,9 @@
 	 */
 	
 	@RequestMapping(method=RequestMethod.POST)
```
Utiliser PostMapping de spring mvc

MawsFr commented on findyourtrashcan-api/src/main/java/fr/femm/findyourtrashcan/security/DataSourceConfig.java:
```diff
@@ -0,0 +1,33 @@
+	private String password;
+
+
+	@Bean(name="dataSource")
```
Constante

MawsFr commented on findyourtrashcan-api/src/main/java/fr/femm/findyourtrashcan/security/DataSourceConfig.java:
```diff
@@ -0,0 +1,33 @@
+	@Bean(name="dataSource")
+	public DriverManagerDataSource dataSource() {
+	    DriverManagerDataSource driverManagerDataSource = new DriverManagerDataSource();
+	    driverManagerDataSource.setDriverClassName("org.postgresql.Driver");
```
Constante

MawsFr commented on findyourtrashcan-api/src/main/java/fr/femm/findyourtrashcan/security/DataSourceConfig.java:
```diff
@@ -0,0 +1,33 @@
+@Configuration
+public class DataSourceConfig {
+	
+	@Value("${spring.datasource.url}")
```
Constante

ExSoldat commented on findyourtrashcan-api/src/main/resources/application-prod.properties:
```diff
@@ -1,7 +1,7 @@
-spring.datasource.url=jdbc:postgresql://localhost:????/*BDDName?useSSL=true
-spring.datasource.username=????
-spring.datasource.password=????
+spring.datasource.url=jdbc:postgresql://localhost:5432/fytcProd?useSSL=true
```
C'est pas un petit peu dangereux de push ces modifications ? Elles ne devraient pas être dans les fichiers qui ne seront jamais modifiés ?

ExSoldat commented on findyourtrashcan-api/src/test/java/fr/femm/findyourtrashcan/controller/FYTCUserControllerTest.java:
```diff
@@ -0,0 +1,137 @@
+
+    }
+    
+    /*
```
Cette partie, on peut la supprimer non ?

MawsFr replied:
oui

Fcornaire replied:
Euh j'avais mis en commentaire car j'avais besoin d'un template pour ça mais ouais je vais le mettre de coté

MawsFr commented on findyourtrashcan-api/src/main/java/fr/femm/findyourtrashcan/security/TokenAuthenticationService.java:
```diff
@@ -0,0 +1,43 @@
+
+class TokenAuthenticationService {
+	static final long EXPIRATIONTIME = 864_000_000; // 10 days
+	static final String SECRET = "ThisIsASecret";
```
ajouter public

Fcornaire replied:
Je pense que c'est pas public exprès car c'est un truc interne

MawsFr commented on findyourtrashcan-api/src/main/java/fr/femm/findyourtrashcan/security/WebSecurityConfig.java:
```diff
@@ -0,0 +1,45 @@
+	    	.csrf().disable()
+	    	.authorizeRequests()
+	    		.antMatchers("/").permitAll()
+	        	.antMatchers(HttpMethod.POST, "/api/user").permitAll()
```
constantes

MawsFr commented on findyourtrashcan-api/src/main/java/fr/femm/findyourtrashcan/security/WebSecurityConfig.java:
```diff
@@ -0,0 +1,45 @@
+	        	.antMatchers(HttpMethod.POST, "/api/user").permitAll()
+	        	.anyRequest().authenticated()
+	    	.and()
+	        .addFilterBefore(new JWTLoginFilter("/api/login", authenticationManager()),
```
constantes

MawsFr commented on findyourtrashcan-api/src/main/java/fr/femm/findyourtrashcan/security/WebSecurityConfig.java:
```diff
@@ -0,0 +1,45 @@
+	  protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+		auth.jdbcAuthentication().dataSource(dataSource)
+		.passwordEncoder(new BCryptPasswordEncoder())
+		.usersByUsernameQuery("SELECT username, password,enabled FROM fytcuser WHERE username=?")
```
constantes

MawsFr commented on findyourtrashcan-api/src/main/java/fr/femm/findyourtrashcan/security/WebSecurityConfig.java:
```diff
@@ -0,0 +1,45 @@
+		auth.jdbcAuthentication().dataSource(dataSource)
+		.passwordEncoder(new BCryptPasswordEncoder())
+		.usersByUsernameQuery("SELECT username, password,enabled FROM fytcuser WHERE username=?")
+		.authoritiesByUsernameQuery("SELECT username" + ", role FROM fytcuser WHERE username=?");
```
constantes

MawsFr commented on findyourtrashcan-api/src/main/resources/application-int.properties:
```diff
@@ -0,0 +1,12 @@
+spring.jpa.database-platform=org.hibernate.dialect.PostgreSQLDialect
+
+# Hibernate ddl auto (create, create-drop, update)
+spring.jpa.hibernate.ddl-auto=create-drop
```
Ne pas droper la base mais laisse le choix au deploy -> modifier le job jenkins
-> pour le moment mettre update

Fcornaire replied:
ok

MawsFr commented on findyourtrashcan-api/src/test/java/fr/femm/findyourtrashcan/controller/FYTCUserControllerTest.java:
```diff
@@ -0,0 +1,137 @@
+
+    @Test
+    public void newUser() throws Exception {
+        mockMvc.perform(post("http://localhost:8080/api/user")
```
constantes on sais jamais le port change non ?

Fcornaire replied:
Ouais pk pas

MawsFr commented on findyourtrashcan-api/src/test/java/fr/femm/findyourtrashcan/service/FYTCUserServiceTest.java:
```diff
@@ -0,0 +1,65 @@
+	@Test
+	public void createUserTest() {
+		//init
+		FYTCUser userToTest = new FYTCUser("maws", "bg", "wo@gmail.com");
```
formater le code ctrl  shift f

MawsFr reviewed (changes requested):
Ajouter des commentaires sur les classes en général, sur leur attribut (pas les getters et setters)

Commit:
<feat> Pull request [Update]
Prise en compte des modifications sur la pull request

Commit:
<feat> Pull request [Role]
-Mise à jour du role utilisé par spring security et par nous (user,admin,agent de la mairie)
-MAJ indentation
